### ISSUE TITLE
Azure: CMake does not find MSBuild options

### ISSUE THREAD

**Issue by xhochy** [conda-forge/conda-forge.github.io#703]
Running cmake in Azure DevOps Pipelines currently fails, e.g. https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=10607&view=logs&jobId=2cc45e14-23e3-52d7-b33a-8c2744410b97&taskId=22e1f59c-6ad0-5e39-c80e-f2cebe3974dd&lineStart=724&lineEnd=725&colStart=1&colEnd=1

The relevant error message is:

```
D:\bld\thrift-cpp_1549275362470\work\build\CMakeFiles\3.13.2\VCTargetsPath.vcxproj(15,2): error MSB4019: The imported project "D:\Microsoft.Cpp.Default.props" was not found. Confirm that the path in the <Import> declaration is correct, and that the file exists on disk.
```

This is probably due to a missing `VCTargetsPath`.

**Comment by mariusvniekerk** [comment]
cc @DavidStaheli @willsmythe

**Comment by mariusvniekerk** [comment]
This looks incredibly similar to https://stackoverflow.com/questions/8497931/msbuild-does-not-contain-a-value-for-the-vctargetspath-property

**Comment by mariusvniekerk** [comment]
I have a small little tester project running here that fetches some of the relevant registry keys

https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=11185

**Comment by mariusvniekerk** [comment]
cc @msarahan @mingwandroid

**Comment by show0k** [comment]
Did you found a workaround ?
When I use cmake -G "Visual Studio 14 2015 Win64", to configure my project, I have the same error 
`      D:\bld\aruco_1562609475662\work\build\CMakeFiles\3.14.5\VCTargetsPath.vcxproj(15,2): error MSB4019: The imported project "D:\Microsoft.Cpp.Default.props" was not found. Confirm that the path in the <Import> declaration is correct, and that the file exists on disk.
`

**Comment by xhochy** [comment]
Known workaround for this is to use `NMake Makefiles` as the CMake generator.

**Comment by jakirkham** [comment]
Yeah using `nmake` on Windows has been useful for several issues. Glad to hear that it works here as well.

**Comment by hadim** [comment]
It looks like this issue is also related: https://github.com/conda-forge/_libarchive_static_for_cph-feedstock/issues/5

`NMake Makefiles` fixed it.

**Comment by xhochy** [comment]
Perferred solution is nowadays to use `Ninja` or `NMake Makefiles JOM` as the generator as `NMake Makefiles` doesn't parallelize.

**Comment by hadim** [comment]
Are those available in the conda forge Windows build tools?

**Comment by xhochy** [comment]
Yes but e.g. for `Ninja` you need to add `ninja` to the `build` requirements.
